components/org.wso2.carbon.identity.workflow.engine/src/main/java/org.wso2.carbon.identity.workflow.engine/DefaultWorkflowEventRequestService.java

```diff
@@ -0,0 +1,151 @@
+import org.wso2.carbon.identity.workflow.mgt.exception.InternalWorkflowException;
+import org.wso2.carbon.identity.workflow.mgt.exception.WorkflowException;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.UUID;
+
+public class DefaultWorkflowEventRequestService implements DefaultWorkflowEventRequest {
+
+    public void addApproversOfRequests(WorkflowRequest request, List<Parameter> parameterList) {
+
+        WorkflowEventRequestDAO workflowEventRequestDAO = new WorkflowEventRequestDAOImpl();
+        String taskId = UUID.randomUUID().toString();
+        String eventId = getRequestId(request);
+        String workflowId = getWorkflowId(request);
+        String approverType = null;
+        String approverName = null;
+        int currentStepValue = getStateOfRequest(eventId, workflowId);
+        if (currentStepValue == 0) {
+            createStatesOfRequest(eventId, workflowId, currentStepValue);
+        }
+        currentStepValue += 1;
+        updateStateOfRequest(eventId, workflowId);
+        List<WorkflowAssociation> associations = getAssociations(request);
+        for (WorkflowAssociation association : associations) {
+            for (Parameter parameter : parameterList) {
+                if (parameter.getParamName().equals(WorkflowEngineConstants.ParameterName.USER_AND_ROLE_STEP)) {
+                    String[] stepName = parameter.getqName().split("-");
+                    int step = Integer.parseInt(stepName[2]);
+                    if (currentStepValue == step) {
+                        approverType = stepName[stepName.length - 1];
+
+                        String approver = parameter.getParamValue();
+                        String[] approvers = approver.split(",");
+                        if (approvers != null) {
+                            List<String> approverList = Collections.singletonList(approver);
+                            String stepValue = WorkflowEngineConstants.ParameterName.USER_AND_ROLE_STEP + "-step-" +
+                                    currentStepValue + "-users";
+                            if (stepValue.equals(parameter.getqName())) {
+                                for (String name : approverList) {
+                                    approverName = name;
+                                }
+                            }
+                            stepValue = WorkflowEngineConstants.ParameterName.USER_AND_ROLE_STEP + "-step-" +
+                                    currentStepValue + "-roles";
+                            if (stepValue.equals(parameter.getqName())) {
+                                for (String name : approverList) {
+                                    approverName = name;
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+        }
+        workflowEventRequestDAO.addApproversOfRequest(taskId, eventId, workflowId, approverType, approverName);
```

Comment:
We need to add approvers if there are multiple approvers for a single step. Include this inside the for loop

Reply:
Addresses with this commit https://github.com/wso2-incubator/identity-workflow-simple/pull/1/commits/0f8bc6df9d488542344585725d992c952c1b640b